shell/browser/browser.cc

```diff
@@ -190,6 +190,9 @@ void Browser::DidFinishLaunching(base::DictionaryValue launch_info) {
   base::FilePath user_data;
   if (base::PathService::Get(chrome::DIR_USER_DATA, &user_data))
     base::CreateDirectoryAndGetError(user_data, nullptr);
+  base::FilePath browser_data;
+  if (base::PathService::Get(DIR_BROWSER_DATA, &browser_data))
+    base::CreateDirectoryAndGetError(browser_data, nullptr);
```

Comment:
Why is this needed? The directory should be created on demand, right?

Reply:
If user sets a custom dir, it is not created on demand. There is code ensuring DIR_USER_DATA  to be created here for the same reason.